lib/practice_exercises.rb

```diff
@@ -1,13 +1,37 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+require 'pry'
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  len = list.length
+  if len == 1 || len == 0
+    return list
+  end
+
+  new_list = []
+  (0..len-1).each do |i|
+    if list[i] != i
+      new_list << list[i]
+    end
```

Comment:
This is only not adding elements if they are not equal to their index, which won't work in all cases.  

Consider:

```ruby
   it "will remove duplicates for even longer arrays " do
      expect(remove_duplicates([1, 1, 1, 1, 1, 2, 2, 3, 3, 4, 4, 4, 4, 4])).must_equal [1, 2, 3, 4]
    end
```